tests/flexmock_test.py

```diff
@@ -856,6 +889,13 @@ def __iter__(self):
         flexmock(Foo).should_receive("__iter__").and_yield(1, 2, 3)
         assert_equal([1, 2, 3], list(Foo()))
 
+    def test_flexmock_should_yield_self_when_iterated(self):
```

Comment:
I wonder what's the point of this behaviour 🤔

Reply:
I'm not sure. Maybe just to make the Mock iterable even if the mocked object is not iterable. Though I don't know what is the use case for this.

Reply:
That sounds hacky, I would be in favour of removing this functionality later, what do you think ?

Reply:
Git blame reveals that this functionality was added for a reason:
https://github.com/has207/flexmock/issues/11
https://github.com/has207/flexmock/issues/74

I could improve the test case based on the examples in these issues.

Reply:
I see, it is a bit weird to always make it iterable though and not just when we mock an iterable class. I will make an issue to investigate this further

Reply:
Created #57

Reply:
Thanks! Let's continue the discussion there. I'll leave this test case for now.